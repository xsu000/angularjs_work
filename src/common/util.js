import { env, envs } from "./global";

/*eslint no-console: ["error", { allow: ["warn", "error", "log"] }] */

export const Ramda = R;

export const Ng = angular;

export const Rx = rxjs;

export const RxOper = rxjs.operators;

export const log = function (msg) {
  if (env === envs.DEV) {
    console.log(msg);
  }
}

export const greetings = 'Hello from.. util!';

/**
 * camelCaseToDash('userId') => "user-id"
 * camelCaseToDash('waitAMoment') => "wait-a-moment"
 * camelCaseToDash('TurboPascal') => "turbo-pascal"
 */
export function camelCaseToDash(str) {
  return str.replace(/([a-zA-Z])(?=[A-Z])/g, '$1-').toLowerCase()
}

/**
 * This version will handle imperfect camel case/pascal case with multiple capital letters in a row
 * @param {string} str
 */
export function camelCaseToDashAdv(str) {
  return str
    .replace(/[^a-zA-Z0-9]+/g, '-')
    .replace(/([A-Z]+)([A-Z][a-z])/g, '$1-$2')
    .replace(/([a-z])([A-Z])/g, '$1-$2')
    .replace(/([0-9])([^0-9])/g, '$1-$2')
    .replace(/([^0-9])([0-9])/g, '$1-$2')
    .replace(/-+/g, '-')
    .toLowerCase();
}

export const generateTemplate = (camelCaseName, properties = []) => {
  const tag = camelCaseToDash(camelCaseName);
  const props = Ramda.join('', Ramda.map(o => {
    return ' ' + o.prop + '="' + o.value + '"';
  }, properties));
  return '<' + tag + props + '></' + tag + '>';
}
