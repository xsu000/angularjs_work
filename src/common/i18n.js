import { Ramda } from "./util";
import enUS from './languages/en-US.json'
import zhCN from './languages/zh-CN.json'

export const LANG_EN_US = 'en-US';
export const LANG_ZH_CN = 'zh-CN';

export const generateLangs = (enUS, zhCN) => {
  let langs = {};
  langs[LANG_EN_US] = enUS;
  langs[LANG_ZH_CN] = zhCN;
  return langs;
}

export const simpleLangs = generateLangs('en', 'cn');

export const COMMON_LANGS = generateLangs(enUS, zhCN);

/**
 * Set the language word JSON and the language preferred.
 * @param {string} lang The language preferred, should be 'en-US', 'zh-CN' etc.
 * @param {object} jsonLangs The i18n language JSON, like {'en-US': {}, 'zh-CN': {}}
 */
export const initI18n = function (lang, jsonLangs) {
  return {
    lang,
    fn: function ($translateProvider) {
      'ngInject';
      // Check https://angular-translate.github.io/docs/#/guide/19_security for sanitize
      $translateProvider.useSanitizeValueStrategy('escape'); // sanitize 
      const allLangs = Ramda.mergeDeepRight(COMMON_LANGS, jsonLangs);
      Ramda.forEachObjIndexed((v, k) => $translateProvider.translations(k, v), allLangs)
      $translateProvider.preferredLanguage(lang);
    }
  }
}
