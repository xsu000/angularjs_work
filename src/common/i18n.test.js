import { Ramda } from "./util";
import { COMMON_LANGS } from './i18n';

describe('I18n test', function () {

  beforeEach(() => {
  });

  // beforeEach(inject((mathService) => {
  // _mathservice = mathService;
  // }));

  test('COMMON_LANGS should exist', function () {
    expect(JSON.stringify(COMMON_LANGS)).toMatchSnapshot();
  })

  test('Deep merge right empty object', function () {
    var actual = Ramda.mergeDeepRight({
      'en-US': { 'abc': 'ABC' },
      'zh-CN': { 'abc': '一二三' }
    }, {
        'en-US': { 'abc': '123' },
        'zh-CN': {},
      });
    expect(actual).toEqual({
      'en-US': { 'abc': '123' },
      'zh-CN': { 'abc': '一二三' },
    });
    expect(actual).toMatchSnapshot();
  });

  test('Deep merge right undefined', function () {
    var actual = Ramda.mergeDeepRight({
      'en-US': { 'abc': 'ABC' },
      'zh-CN': { 'abc': '一二三' }
    }, {
        'zh-CN': undefined,
      });
    expect(actual).toEqual({
      'en-US': { 'abc': 'ABC' },
      'zh-CN': undefined,
    });
  });

  test('Deep merge right null', function () {
    var actual = Ramda.mergeDeepRight({
      'en-US': { 'abc': 'ABC' },
      'zh-CN': { 'abc': '一二三' }
    }, {
        'zh-CN': null,
      });
    expect(actual).toEqual({
      'en-US': { 'abc': 'ABC' },
      'zh-CN': null,
    });
  });

});