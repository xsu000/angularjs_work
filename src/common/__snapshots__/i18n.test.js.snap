// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`I18n test COMMON_LANGS should exist 1`] = `"{\\"en-US\\":{\\"PAGE_TITLE\\":\\"AngularJS ES6 multi page demo\\"},\\"zh-CN\\":{\\"PAGE_TITLE\\":\\"AngularJS ES6 多页面 demo\\"}}"`;

exports[`I18n test Deep merge right empty object 1`] = `
Object {
  "en-US": Object {
    "abc": "123",
  },
  "zh-CN": Object {
    "abc": "一二三",
  },
}
`;
